### ISSUE TITLE
Racial tech is actually global

### ISSUE THREAD

**Issue by marhawkman** [ekolis/stars-nova#7]
**Describe the bug**
Items that should be racial tech are available to all races

**To Reproduce**
In this example I managed to simultaneously research Croby Sharmor(an Inner Strength tech) and Shadow Shield(Super Stealth) simultaneously, but it seems to apply to all racial tech as in the menus I also have Heavy mines(Space Demolition) and Super Mass drivers(Packet Physics)
![racetechc](https://user-images.githubusercontent.com/64557483/80723034-b2223c80-8ac5-11ea-8644-c94191236659.png)

**Expected behavior**
racial tech is for those races only

**Actual behavior**
everyone seems to get racial tech(Note only components, not ship hulls)

**Screenshots**

**Additional context**

**Comment by marhawkman** [comment]
on second thought I think I am getting some of the racial tech hulls too, it might be certain races affected.  IE everyone gets Inner Strength tech, rather than components.

**Comment by marhawkman** [comment]
Ok, I looked again and realized I also have techs that require minor traits.

In my current test game I'm playing Inner Strength I can use in designs: Depleted Neutronium(Super Stealth), Orbital adjustors(Claim Adjustor), Heavy Mines(Space Demolition), Stargate Any/300(Interstellar Traveller). Also I don't have advanced remote mining or improved starbases but have those techs available anyways.

**Comment by ekolis** [comment]
Yeah, I think this and #18 are related - there's a bug in the tech requirements code that causes it to think you are unlocking tech you don't actually have, and another bug there where it completely ignores racial traits. I'd fix these bugs but I created a unit test because the logic is kind of tricky and I want to get it right, and the unit tests won't run for some reason...
